### PR TITLE
[New Concept Entries] Visual Design, Typography, Design Pattern, Design System, Design Component, Style Guide + [Entry Updates] Information Architecture Content + Old Entry Attribution

### PR DESCRIPTION
**Pull request by notalyce**
### Description

[New Concept Entries]
Visual Design
Typography
Design Pattern
Design System
Design Component
Style Guide

[Entry Updates]
Information Architecture [Content update]
Old entries updated with 'design' tag for attribution

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/uiux/concepts/design-component/design-component.md
Comment (lucwhite) [inline content/uiux/concepts/design-component/design-component.md:3]: This is for SEO and is the little blurb that shows up in google under the link. It'd probably be good to fit in mention of design systems and a few examples (buttons, icons, form fields, etc.)

**Review by lucwhite — APPROVED** [review]
Looks good to me! Will wait on the next review from the maintainers :D

File: content/uiux/concepts/design-component/design-component.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-component/design-component.md:15]:
```suggestion
**Design components** are specific examples of design elements that might exist within a given context. They can be thought of like building blocks for design, which are often documented within a component library as part of a larger [design system](https://www.codecademy.com/resources/docs/uiux/design-system). Examples of design components could include elements such as buttons, headings, icons, form fields, and menus.
```

File: content/uiux/concepts/design-component/design-component.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-component/design-component.md:17]:
```suggestion
To dive deeper into one example, a [button](https://www.codecademy.com/resources/docs/uiux/button) designed in accordance with the [style guide](https://www.codecademy.com/resources/docs/uiux/style-guide) for a specific website or brand would qualify as a component. Buttons are clickable elements often used to highlight certain calls-to-action or links on a website, and are a common design pattern used across all kinds of websites and applications.
```

File: content/uiux/concepts/design-pattern/design-pattern.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-pattern/design-pattern.md:15]:
```suggestion
**Design patterns** are standardized, widely adopted solutions to common design problems. Design patterns tend to be more flexible than [design components](https://www.codecademy.com/resources/docs/uiux/design-component), and can include specific modules, flows, or behaviors.
```

File: content/uiux/concepts/design-pattern/design-pattern.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-pattern/design-pattern.md:17]:
```suggestion
For example, if a design includes a progress indicator, it's not necessary to reinvent the wheel. Instead, designers are likely to select from common patterns and adjust them to their context.
```

File: content/uiux/concepts/design-pattern/design-pattern.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-pattern/design-pattern.md:21]:
```suggestion
Once a designer has a sense for what types of actions a user will need to take within an application or interface, design patterns from competitive analysis and pattern libraries can be used as a reference for building out user flows. Competitive analysis is the process of examining designs with related functionality as part of the design research process. Pattern libraries are websites that collect and define different UX and UI patterns.
```

File: content/uiux/concepts/design-pattern/design-pattern.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-pattern/design-pattern.md:25]:
```suggestion
Dark patterns are tricky or deceptive design patterns used in websites and apps that make users do things that they didn't intend to. While dark design patterns may lead to short-term business gains, they ultimately diminish trust over time and do a disservice to the people using our designs.
```

File: content/uiux/concepts/design-pattern/design-pattern.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-pattern/design-pattern.md:31]:
```suggestion
A date picker is a common design pattern used on many different types of websites and applications. This design pattern should allow users to easily select a specific date or date range. Some common formats include calendars, dropdowns, text fields, or hybrid patterns.
```

File: content/uiux/concepts/design-pattern/design-pattern.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-pattern/design-pattern.md:37]:
```suggestion
Search results are a list of results displayed after a search form is completed and submitted. Search results can follow a variety of different design patterns. It’s fairly common for search results to be showed as a list view, but designers could also potentially use a card view or table view. Designers should also consider what information should be displayed about each result, the order of results, what filters to make available to users, the desired next steps after search, how users will scroll through results, and whether specific results should be differentiated visually.
```

File: content/uiux/concepts/design-system/design-system.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-system/design-system.md:15]:
```suggestion
**Design systems** are repositories of reusable [design components](https://www.codecademy.com/resources/docs/uiux/design-component) and [design patterns](https://www.codecademy.com/resources/docs/uiux/design-pattern) that can be used across a team or organization to ensure consistency across all design output.
```

File: content/uiux/concepts/design-system/design-system.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-system/design-system.md:22]:
```suggestion
- **Component library:** Component libraries are repositories of reusable [design components](ttps://www.codecademy.com/resources/docs/uiux/design-component) that designers can easily copy into new designs. For example, if a designer needs to add a button to an input form, they could copy a component from the component library.
```

File: content/uiux/concepts/design-system/design-system.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-system/design-system.md:23]:
```suggestion
- **Pattern library:** Pattern libraries are repositories of reusable [design patterns](https://www.codecademy.com/resources/docs/uiux/design-pattern) which focus on layouts that could include multiple components and focus on content structure. For example, a login pattern might include multiple components such as a header, text input fields, a button, and potential error or success indicators.
```

File: content/uiux/concepts/design-system/design-system.md
Comment (SSwiniarski) [inline content/uiux/concepts/design-system/design-system.md:27]:
```suggestion
There are many ways to document a design system, and different organizations will have their own approaches. Separate sections of the system may live within different tools. For example, some organizations might maintain an internal wiki with their style guide, while their component library is maintained in [digital design software](https://www.codecademy.com/resources/docs/uiux/design-software) like Figma or Sketch for easy duplication into new design prototypes. One of the challenges of design systems is ensuring that they are maintained. Both building and updating the system requires time and effort.
```

File: content/uiux/concepts/information-architecture/information-architecture.md
Comment (SSwiniarski) [inline content/uiux/concepts/information-architecture/information-architecture.md:19]:
```suggestion
Effective IA allows users to complete tasks with the least effort required. [Site maps](https://www.codecademy.com/resources/docs/uiux/site-map) are a visual representation of IA and give context to how content is tiered. Although IA isn’t visually presented to users, content should be grouped and structured in a way that makes logical sense to maximize usability.
```

File: content/uiux/concepts/information-architecture/information-architecture.md
Comment (SSwiniarski) [inline content/uiux/concepts/information-architecture/information-architecture.md:23]:
```suggestion
Information architecture is heavily influenced by mental models. The designer model is the mental image in the head of the designer. The user model is the mental image in the head of the user. In an ideal world, the designer’s model and user’s model should be exactly the same - meaning the user will know exactly how to use a product. The system model is the actual image conveyed by the product and written material. The better the system model is, the closer the designer model and user model will be, which in turn, makes the product easier to understand.
```

File: content/uiux/concepts/style-guide/style-guide.md
Comment (SSwiniarski) [inline content/uiux/concepts/style-guide/style-guide.md:15]:
```suggestion
In UI and UX design, a **style guide** documents guidelines to create consistency across visual design and branding elements, such as approved colors, logo treatments, and typefaces. Some style guides may include further guidance on written content, UX copy, or [interaction design](https://www.codecademy.com/resources/docs/uiux/interaction-design). A brand’s style guide is often documented as part of a larger [design system](https://www.codecademy.com/resources/docs/uiux/design-system).
```

File: content/uiux/concepts/typography/typography.md
Comment (SSwiniarski) [inline content/uiux/concepts/typography/typography.md:21]:
```suggestion
In discussing type, the terms “font” and “typeface” are commonly encountered. In common usage today, the term **font** is broadly used to describe a set of characters of the same style. Historically, **typeface** referred to an entire family of fonts, such as Helvetica, Futura, or Times New Roman, while font referred to a specific size and weight of that typeface, such as “Helvetica Bold 12 pt.” Today, the terms font and typeface are often used interchangeably.
```

File: content/uiux/concepts/visual-design/visual-design.md
Comment (SSwiniarski) [inline content/uiux/concepts/visual-design/visual-design.md:15]:
```suggestion
**Visual design** is a design discipline that aims to develop a strategic aesthetic for digital products. This aesthetic must organize [visual design elements](https://www.codecademy.com/resources/docs/uiux/visual-design-elements) such as line, shape, texture, space, form, and typography into a cohesive whole and support usability. Additionally, visual designers seek to create interfaces that are visually appealing.
```

File: content/uiux/concepts/visual-design/visual-design.md
Comment (SSwiniarski) [inline content/uiux/concepts/visual-design/visual-design.md:17]:
```suggestion
Visual design is derived from graphic design and [UI design](https://www.codecademy.com/resources/docs/uiux/ui-design). In the digital space, graphic designers may work on visuals related to a product’s marketing and branding. By contrast, visual designers and UI designers are responsible for the look of the product interface. Visual designers focus on the overall aesthetic of a product and may be in charge of building and maintaining a brand’s [design system](https://www.codecademy.com/resources/docs/uiux/design-system).
```

File: content/uiux/concepts/visual-design/visual-design.md
Comment (SSwiniarski) [inline content/uiux/concepts/visual-design/visual-design.md:23]:
```suggestion
While visual design is essential, not all users rely on visual information to use technology. People with vision impairments may rely on sound, haptics, and visual input when interacting with digital products. Thus, [accessibility](https://www.codecademy.com/resources/docs/uiux/accessibility) should be a priority for designing inclusive products.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@notalyce I did a first review. This all looks good, except we generally don't use bold to set off terms in Docs. That's where the majorirty of my edits come from.

**Comment by lucwhite** [comment]
Per @aherman91 -- we'll stick with the bold in these entries, and may start to implement that across the board as it better matches our content standards. I'll mark those suggestions as outdated.

cc @SSwiniarski @notalyce

File: content/uiux/concepts/design-system/design-system.md
Comment (lucwhite) [reply content/uiux/concepts/design-system/design-system.md:22]:
```suggestion
- **Component library:** Component libraries are repositories of reusable **[design components](ttps://www.codecademy.com/resources/docs/uiux/design-component)** that designers can easily copy into new designs. For example, if a designer needs to add a button to an input form, they could copy a component from the component library.
```

File: content/uiux/concepts/visual-design/visual-design.md
Comment (lucwhite) [reply content/uiux/concepts/visual-design/visual-design.md:17]: I think this one is ok, graphic design and UI design are unlikely to be new terms for learners.

File: content/uiux/concepts/design-system/design-system.md
Comment (lucwhite) [reply content/uiux/concepts/design-system/design-system.md:27]: I think this one is ok, digital design software is pretty self explanatory and we provide examples right there.

File: content/uiux/concepts/style-guide/style-guide.md
Comment (lucwhite) [reply content/uiux/concepts/style-guide/style-guide.md:15]: I think this suggestion is ok as well.

File: content/uiux/concepts/typography/typography.md
Comment (SSwiniarski) [reply content/uiux/concepts/typography/typography.md:21]: @notalyce It's been a standard in Docs from the beginning to avoid direct address to the reader. We avoid first- and second-person pronouns.

Comment (lucwhite) [reply content/uiux/concepts/typography/typography.md:21]: Yeah, Steven's correct here, on docs we use passive voice in these situations to ensure we're just speaking about the facts and there're no assumed actors. 

It's a spot where docs standards diverge from content standards.

**Comment by lucwhite** [comment]
@SSwiniarski any other changes necessary here?

**Comment by SSwiniarski** [comment]
Looks good to me. Should we have a second review?

**Review by SSwiniarski — APPROVED** [review]

File: content/uiux/concepts/accessibility/accessibility.md
Comment (yangc95) [inline content/uiux/concepts/accessibility/accessibility.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/context-of-use/context-of-use.md
Comment (yangc95) [inline content/uiux/concepts/context-of-use/context-of-use.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/convergent-thinking/convergent-thinking.md
Comment (yangc95) [inline content/uiux/concepts/convergent-thinking/convergent-thinking.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/design-component/design-component.md
Comment (yangc95) [inline content/uiux/concepts/design-component/design-component.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/design-methodologies/design-methodologies.md
Comment (yangc95) [inline content/uiux/concepts/design-methodologies/design-methodologies.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/design-pattern/design-pattern.md
Comment (yangc95) [inline content/uiux/concepts/design-pattern/design-pattern.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/design-system/design-system.md
Comment (yangc95) [inline content/uiux/concepts/design-system/design-system.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/design-thinking/design-thinking.md
Comment (yangc95) [inline content/uiux/concepts/design-thinking/design-thinking.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/design/design.md
Comment (yangc95) [inline content/uiux/concepts/design/design.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/divergent-thinking/divergent-thinking.md
Comment (yangc95) [inline content/uiux/concepts/divergent-thinking/divergent-thinking.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/double-diamond-diagram/double-diamond-diagram.md
Comment (yangc95) [inline content/uiux/concepts/double-diamond-diagram/double-diamond-diagram.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/ethnography-field-studies/ethnography-field-studies.md
Comment (yangc95) [inline content/uiux/concepts/ethnography-field-studies/ethnography-field-studies.md:9]:
```suggestion
  - 'Design'
```

File: content/uiux/concepts/design-system/design-system.md
Comment (yangc95) [inline content/uiux/concepts/design-system/design-system.md:22]:
```suggestion
- **Component library:** Component libraries are repositories of reusable [design components](https://www.codecademy.com/resources/docs/uiux/design-component) that designers can easily copy into new designs. For example, if a designer needs to add a button to an input form, they could copy a component from the component library.
```

**Review by yangc95 — APPROVED** [review]

**Pull request merged**
